avaje-config/src/main/java/io/avaje/config/InitialLoader.java

```diff
@@ -272,6 +285,23 @@ boolean loadWithExtensionCheck(String fileName) {
     }
   }
 
+  private boolean loadURI(String uriPath) {
+    var uri = URI.create(uriPath);
+    final var scheme = uri.getScheme();
+    var loader = uriLoaders.get(scheme);
+    if (loader != null) {
+      loader.load(uri, parsers).forEach((k, v) -> loadContext.put(k, v, "uri scheme " + scheme));
```

Comment:
Ideally the URI itself would be used for the source part. e.g.:

```java
loadContext.put(k, v, "uri: " + redact(uri));
```

The `redact` function would check for `password`, `key`, `apikey`, `secret` and if the URI contains it do `"<REDACTED for " + scheme +">"` instead of the `URI`.

Then again your solution does not need that redact function 😄 so maybe we just stick to  what you have.

Reply:
I like the elegant simplicity